src/backends/gr.jl

```diff
@@ -1493,18 +1487,14 @@ function gr_draw_axes(sp, viewport_plotarea)
         x_bg, y_bg = RecipesPipeline.unzip(GR.wc3towc.(area_x, area_y, area_z))
         GR.fillarea(x_bg, y_bg)
 
-        for letter in (:x, :y, :z)
-            gr_draw_axis_3d(sp, letter, viewport_plotarea)
-        end
+        foreach(letter -> gr_draw_axis_3d(sp, letter, viewport_plotarea), (:x, :y, :z))
```

Comment:
I'm not very comfortable with the amount of closures you are introducing lately. I hope, you tested, that you don't run into https://github.com/JuliaLang/julia/issues/15276
it would be more robust to wrap thos in `let` block for safety.

Reply:
I'll think about it. To me atm, being concise is more important. Also, this is not a critical function called a thousand times per plot. But I get your point.

Reply:
Its not this specific place. You do that all over the codebase, which is fine if you made sure this does not impact performance, otherwise it will just make work to revert it where it did.

> To me atm, being concise is more important.

Users usually disagree about that ^^

Reply:
**master**
```julia
julia> using Plots; @time plot(1:2) |> display
  6.766723 seconds (713.40 k allocations: 35.765 MiB, 0.65% gc time, 95.10% compilation time: 15% of which was recompilation)
```

**1.35.0**
```julia
julia> using Plots; @time plot(1:2) |> display
  6.649556 seconds (714.62 k allocations: 35.905 MiB, 0.67% gc time, 95.05% compilation time: 15% of which was recompilation)
```

Reply:
That case is covered by the benchmarks ci job. But there is is enough code that was changed that is not covered by this.
I'm just saying you should test _the specific function you are changing_ before merge, whether stylistic changes impact performance.

Reply:
I hear that, but there is theory and practice.
I'm also saying that having good development practices involves good coverage, which I'm trying hard to improve here. I'm not even talking about the backend bugs, ...
Performance discussions should always happen last.

Reply:
> I'm also saying that having good development practices involves good coverage, which I'm trying hard to improve here.

I see that. And in general, I appreciate that. But potentially making existing code worse just to have a better score on some debatable development metric is the wrong way to go.

Reply:
IMO, there is nothing to be discussed when dealing with widely used codebases such as `Plots`.
Robustness **should** prevail over performance.
There will be a time where we will focus on reducing allocations, which is `julia`'s weak point these days.
But first robustness through coverage (which doesn't mean I will write sloppy code). Non tested code is bound to fail.

Reply:
Its not increasing robustness if you make functional equivalent changes to reduce line count.
Either that code paths gets tested or not. It's gaming numbers.
I mean sometimes it is beneficial to use `map` instead of a comprehension and sometimes its not.